azkustodata/query/v2/iteartive_dataset.go

```diff
@@ -43,6 +43,7 @@ func NewIterativeDataset(ctx context.Context, r io.ReadCloser, capacity int) (qu
 	}
 
 	go func() {
+		defer d.reader.Close()
```

Comment:
i think this reader leaks if prepareReadBuffer fails

Reply:
Great catch!